src/riscv/lib/src/machine_state/memory/protection.rs

```diff
@@ -73,11 +73,11 @@ impl<const PAGES: usize, M: ManagerBase> PagePermissions<PAGES, M> {
             return true;
         }
 
-        let address = address as usize;
-
         // Extract the page index range from using the start and end addresses
-        let start_page = address >> super::OFFSET_BITS.get();
-        let end_page = address.wrapping_add(length).wrapping_sub(1) >> super::OFFSET_BITS.get();
+        let start_page = super::address_to_page_index(address);
```

Comment:
nit: free to resolve

```suggestion
        let start_page = address_to_page_index(address);
```
(and in the other places)

Reply:
done